### ISSUE TITLE
JS error: TypeError: Cannot read property 'status' of undefined

### ISSUE THREAD

**Issue by thebluebug** [whitecube/nova-flexible-content#57]
Previously we were getting this error (https://github.com/whitecube/nova-flexible-content/issues/55), so I decided to update from 0.1.7 to 0.1.9. That error is gone, but now I'm getting a completely different error when trying to create/update. The create/update button won't let me submit. Here's the JS error:

```
TypeError: Cannot read property 'status' of undefined
    at p.<anonymous> (app.js?id=b8265367338097693595:formatted:13231)
    at y (app.js?id=b8265367338097693595:formatted:16860)
    at Generator._invoke (app.js?id=b8265367338097693595:formatted:16838)
    at Generator.e.<computed> [as next] (app.js?id=b8265367338097693595:formatted:16875)
    at o (app.js?id=b8265367338097693595:formatted:22898)
    at app.js?id=b8265367338097693595:formatted:22910
    at new Promise (<anonymous>)
    at new t (app.js?id=b8265367338097693595:formatted:25246)
    at p.<anonymous> (app.js?id=b8265367338097693595:formatted:22895)
    at p.<anonymous> (app.js?id=b8265367338097693595:formatted:13239)
It @ vendor.js?id=da751a53228a35fab7c4:formatted:39444

```
The culprit might be R64's Nova Fields again (if that's the case, I'll open a ticket with them), but we need to wrap Flexible content and a few other fields in this R64 JSON field for our data needs:
```
            JSON::make('Details', [
                Text::make('Test', 'test'),
                Text::make('Test 2', 'test2'),
                Flexible::make('Student Choice Selections', config('magento.gp_bundle_student_choices'))
                    ->button('Add Student Choice Item')
                    ->addLayout('Student Choice Selections', 'student_choice_selections', [
                        Text::make('Choice Description', config('magento.gp_bundle_student_products.selection_title'))
                            ->fieldClasses('choice-description-field px-8 py-6')
                            ->rules('required'),
               Text::make('Test 3', 'test3'),
               Text::make('Test 4', 'test4'), 
            ], 'data')->fieldClasses('w-full')
                ->labelClasses('w-1/6 px-8 py-6 hidden')
                ->hideFromIndex(),
```

Wrapping it with another Flexible field instead of this JSON field isn't going to work for us.

**Comment by thebluebug** [comment]
@Nyratas ?

**Comment by Nyratas** [comment]
Hi @thebluebug,

We just released v0.1.10. Could you check if this issue is still relevant?

Do not forget to `php artisan nova:publish` & hard-reload (`cmd + shift + R`) Nova in your browser before submitting test data.

**Comment by thebluebug** [comment]
@Nyratas Updated to v0.1.10. Issue remains.

**Comment by thebluebug** [comment]
@Nyratas Updated to v0.1.11. Issue remains. I also posted about this issue over here: https://github.com/64robots/nova-fields/issues/118

**Comment by thebluebug** [comment]
Any insights into how I can resolve this?

**Comment by thebluebug** [comment]
@Nyratas I figured out where it fails and gives the unhelpful JS error from above.
FormField.vue line 138:
`if(formData.has('___nova_flexible_content_fields')) {`

If I comment out line 123 (`this.appendFieldAttribute(formData, this.field.attribute);`), the error goes away, and I can press the Create button, but then I get the other error again:
`[2019-09-05 17:44:51] localhost.ERROR: Unable to parse incoming Flexible content, data should be an array. {"userId":2,"exception":"[object] (Exception(code: 0): Unable to parse incoming Flexible content, data should be an array. at /var/www/html/vendor/whitecube/nova-flexible-content/src/Flexible.php:309)`

And then my required fields within Flexible don't fail validation as they should... so any insights would be helpful.

**Comment by Nyratas** [comment]
Hi @thebluebug,

Thanks for the research. The `___nova_flexible_content_fields` is required because it's the only way the FlexibleContent is able to parse the incoming request data for validation. This means that removing the `appendFieldAttribute` method is not the solution.

But it is interesting that `formData.has('___nova_flexible_content_fields')` is responsible for the error you're getting. Could you share the result of `console.log(formData)` before the `if` statement on line 138? Thanks!

**Comment by thebluebug** [comment]
@Nyratas Right, I know it's not the solution. I forgot to mention that. I was just showing where the actual problem might be. I should've included what's in `formData` since I was comparing it to the last version that works for us (v0.1.7) while I was debugging this.

Anyway, here it is with v0.1.12 (it's missing `data[student_choices]`):
```
n
attribute: "data"
obj: data[bal_pdp_mktg]: "Test"
    data[bundle_default_products]: "[{"sku":"A0009728A0000","name":"Lightweight Class of '20 Adjustable Hat (Black)","price":25,"category":"Apparel-Class Of 2020 Hats","vgt_id":0,"originalIndex":0,"vgtSelected":true}]"
    data[description]: "Test"
    data[hero_image]: ""
    data[short_description]: "Test"
__proto__: Object
__proto__: Object
```

In v0.1.7 the same as above but with `data[student_choices]` appended after `data[short_description]` (below is the Flexible field with a few other fields and the R64 JSON field wraps `bal_pdp_mktg`, `bundle_default_products`, `description`, `hero_image`, `short_description`, and `student_choices`):
```
data[student_choices]: "[{"layout":"student_choice_selections","key":"nqrto4tphgpo7iay2ka2iq-student_choice_selections","attributes":{"nqrto4tphgpo7iay2ka2iq-student_choice_selections__title":"","nqrto4tphgpo7iay2ka2iq-student_choice_selections__max_allowed":"","nqrto4tphgpo7iay2ka2iq-student_choice_selections__products":"[]"}}]"
```

**Comment by XternalSoft** [comment]
Any news ?
This is a critical bug... I can't save anymore  :'(

**Comment by thebluebug** [comment]
@XternalSoft I can't either. @Nyratas Any thoughts on how to resolve this?

**Comment by XternalSoft** [comment]
@thebluebug !! I founded the solution (for me, expected for you too).

It's was an another nova package thats make "conflict". 
I just removed "tanmuhittin/nova-scroll-top" from my composer.json and now i can save in Flexible Content !! 

I'm so happy !! lol

**Comment by thebluebug** [comment]
@XternalSoft We aren't using that nova package. I'm glad you were able to solve!

**Comment by XternalSoft** [comment]
Ok, but maybe an another package for you ?

**Comment by thebluebug** [comment]
@XternalSoft No. As a workaround, we wound up locking to the version of Flexible that worked for us (v0.1.7) and then did server-side validation to check the two required fields (a Text and Select field) wrapped within the Flexible field. It throws a Toasted error message if one or both fields are not filled out when you click on the Create button.

**Comment by XternalSoft** [comment]
:confused:
